### PR TITLE
HP-366 Get and show Verified user data

### PR DESCRIPTION
**Pull request by NikoHelle**
Updated query to get the data and component to render it. Result is just plain text, no styling.

Also improved Example UI's errors rendering. No it is easier to find out why errors occur without checking console or logs.

### CODE REVIEW

**Comment by NikoHelle** [comment]
> If "Auto formatting made a lot of changes", then commit those to their own commit.

That file contains the change that does what the commit is all about! I cant save it without auto formatting! Should I open it in some text editor, make the changes and commit. Then open the file in actual code editor, let it auto format and commit again?

**Review by akikoskinen — COMMENTED** [review]
I noticed that if going to the Profile tab, but the logged in user doesn't have a profile (i.e. server returns `null` as `profile`), then the view keeps saying "Loading..." (or "Ladataan..."). I admit this is a bit unusual situation, but perhaps it could be handled too?

**Comment by akikoskinen** [comment]
> > If "Auto formatting made a lot of changes", then commit those to their own commit.
> 
> That file contains the change that does what the commit is all about! I cant save it without auto formatting! Should I open it in some text editor, make the changes and commit. Then open the file in actual code editor, let it auto format and commit again?

There are many ways to split the edits to multiple commits. `git commit -a` isn't the only available way to do commits. With `git add -i` you can add parts of the changes (even edit the patches while adding). Select "5" (patch) and follow the instructions.

**Comment by NikoHelle** [comment]
> > > If "Auto formatting made a lot of changes", then commit those to their own commit.
> > 
> > 
> > That file contains the change that does what the commit is all about! I cant save it without auto formatting! Should I open it in some text editor, make the changes and commit. Then open the file in actual code editor, let it auto format and commit again?
> 
> There are many ways to split the edits to multiple commits. `git commit -a` isn't the only available way to do commits. With `git add -i` you can add parts of the changes (even edit the patches while adding). Select "5" (patch) and follow the instructions. Build the staging area (or index as it's also called), checking how looks, and once you are happy with it, `git commit`.

I think this is a good example following the rule book vs common sense. Is the gain here worth the effort?

Auto formatting while saving is not indicated in any way while coding. Unless you spot changes in real time after cmd+s.  When you maybe days later make a PR, you see there are new spaces added to some lines in one commit. "There was some auto formatting", you think. "Lets add a comment to the commit, so reviewer understands what happened".  No, you should find the commit hash, rebase to that, remember what has your additions and what was auto format, split the commit, add messages, continue rebase and push. Just to get spaces out from one commit.

If I open an old file made with different linter rules and rule for line's max length has changed. You make 20 new lines of code. Two lines are changed by the linter and they split into two lines. Is another commit needed for those too?

What does the reviewer gain?

**Comment by NikoHelle** [comment]
> I noticed that if going to the Profile tab, but the logged in user doesn't have a profile (i.e. server returns `null` as `profile`), then the view keeps saying "Loading..." (or "Ladataan..."). I admit this is a bit unusual situation, but perhaps it could be handled too?

Thx, didnt see that with my users. Fixed in new commit.

File: src/components/Profile.tsx [hunk 1]
```diff
@@ -4,14 +4,40 @@ import { ApiAccessTokenActions } from '../client/hooks';
...
 import { ApiAccessTokenContext } from './ApiAccessTokenProvider';
 import styles from './styles.module.css';
+import { AnyObject } from '../common';
+
+const nodeToJSON = (node: AnyObject): AnyObject | AnyObject[] => {
+  if (Array.isArray(node.edges)) {
+    return node.edges.map(edge => nodeToJSON(edge.node) as AnyObject);
+  }
+  // eslint-disable-next-line no-underscore-dangle
+  if (node.__typename === 'VerifiedPersonalInformationNode') {
```
[the remark this answers is not in the document]
Comment (NikoHelle) [reply src/components/Profile.tsx]: Good idea. Whitelisted that name. Rule not disabled, because "_" should not be used in general.

**Comment by akikoskinen** [comment]
> > > > If "Auto formatting made a lot of changes", then commit those to their own commit.
> > > 
> > > 
> > > That file contains the change that does what the commit is all about! I cant save it without auto formatting! Should I open it in some text editor, make the changes and commit. Then open the file in actual code editor, let it auto format and commit again?
> > 
> > 
> > There are many ways to split the edits to multiple commits. `git commit -a` isn't the only available way to do commits. With `git add -i` you can add parts of the changes (even edit the patches while adding). Select "5" (patch) and follow the instructions. Build the staging area (or index as it's also called), checking how looks, and once you are happy with it, `git commit`.
> 
> I think this is a good example following the rule book vs common sense. Is the gain here worth the effort?
> 
> Auto formatting while saving is not indicated in any way while coding. Unless you spot changes in real time after cmd+s. When you maybe days later make a PR, you see there are new spaces added to some lines in one commit. "There was some auto formatting", you think. "Lets add a comment to the commit, so reviewer understands what happened". No, you should find the commit hash, rebase to that, remember what has your additions and what was auto format, split the commit, add messages, continue rebase and push. Just to get spaces out from one commit.
> 
> If I open an old file made with different linter rules and rule for line's max length has changed. You make 20 new lines of code. Two lines of the old code are changed by the linter and they split into two lines. Is another commit needed for those too?
> 
> What does the reviewer gain?

I guess the only right answer is "it depends". If there are one or two line changes made by a linter, then I would say they are fine within another commit. If the linter changes 20 lines (e.g. reformats a whole file), and the actual code change is small compared to that, then separate commits.

When making commits (most of them probably temporary) and rewriting previous commits often, you get really fast with it. Then you don't think any more if you should better the code, you'll just go and do it. It's like a chef caring about those fine final details in the dishes they make, over and over again.

Commits aren't made solely or not even mainly for reviewers. They are made for the future developers reading the code, including the original author.

Another way of thinking about it: what if some code change will be later judged as incorrect (the decision to do that feature is canceled or whatever the reason), is there a commit or multiple commits that can be simply reverted? And while doing so no code changes, that **shouldn't** have been reverted, got reverted.

**Comment by NikoHelle** [comment]
> Another way of thinking about it: what if some code change will be later judged as incorrect (the decision to do that feature is canceled or whatever the reason), is there a commit or multiple commits that can be simply reverted? And while doing so no code changes, that **shouldn't** have been reverted, got reverted.

This is a great argument for creating two commits.

**Review by akikoskinen — APPROVED** [review]

**Pull request merged**